compiler/resolution/preFold.cpp

```diff
@@ -2238,6 +2238,45 @@ static Expr* preFoldNamed(CallExpr* call) {
 
       retval = unrollHetTupleLoop(call, tupExpr, iterType);
     }
+
+  } else if (call->numActuals() == 1) {
+    // Implement the common case of a boolean argument here,
+    // to avoid full-blown call resolution.
+    if (SymExpr* argSE = toSymExpr(call->get(1))) {
+      Symbol* argSym = argSE->symbol();
+      Type* argType = argSym->type;
+
+      if (call->isNamed("_cond_test") || call->isNamed("isTrue")) {
+        if (argType == dtBool) {
+          // use the argument directly
+          retval = argSE->remove();
+
+        } else if (argType == dtBool->refType) {
+          // dereference it so later passes get a value
+          Expr* stmt = call->getStmtExpr();
+          VarSymbol* repl = newTempConst(dtBool);
+          stmt->insertBefore(new DefExpr(repl));
+          stmt->insertBefore("'move'(%S,'deref'(%E))", repl, argSE->remove());
```

Comment:
That's fancy, I wasn't aware we could do that

Reply:
It does mean I should search for 'move' and PRIM_MOVE if I ever modify what PRIM_MOVE does in the future, though . . .